util/src/main/java/com/redhat/lightblue/util/metrics/DropwizardRequestMetrics.java

```diff
@@ -0,0 +1,138 @@
+    
+    private final Logger LOGGER = LoggerFactory.getLogger(DropwizardRequestMetrics.class);
+
+    private static final String API = "api";
+
+    private final MetricRegistry metricsRegistry;
+    
+    private final String REGEX = "[-@#!$%^&*()_+|~={}:;'<>?,\"\\s]";
+
+    public DropwizardRequestMetrics(MetricRegistry metricRegistry) {
+        metricsRegistry = metricRegistry;
+    }
+
+    /**
+     * Create exception namespace for metric reporting based on exception name
+     * 
+     */
+    private static String errorNamespace(String metricNamespace, Throwable exception) {
+        Class<? extends Throwable> actualExceptionClass = unravelReflectionExceptions(exception);
+        return name(metricNamespace, "requests", "exception", actualExceptionClass.getSimpleName());
+    }
+
+    /**
+     * Get to the cause we actually care about in case the bubbled up exception is a
+     * higher level framework exception that encapsulates the stuff we really care
+     * about.
+     * 
+     */
+    private static Class<? extends Throwable> unravelReflectionExceptions(Throwable e) {
+        while (e.getCause() != null
+                && (e instanceof UndeclaredThrowableException || e instanceof InvocationTargetException)) {
+            e = e.getCause();
+        }
+        return e.getClass();
+    }
+
+    @Override
+    public Context startEntityRequest(String operation, String entity, String version) {
+        return new DropwizardContext(name(API, operation, entity, version));
+    }
+
+    @Override
+    public Context startStreamingEntityRequest(String operation, String entity, String version) {
+        return new DropwizardContext(name(API, "stream", operation, entity, version));
+    }
+    
+    @Override
+    public Context startLockRequest(String lockOperation, String domain) {
+        return new DropwizardContext(name(API, "lock", domain, lockOperation));
+    }
+
+    @Override
+    public Context startBulkRequest() {
+        return new DropwizardContext(name(API, "bulk"));
+    }
+
+    public class DropwizardContext implements Context {
+        
+        private final String metricNamespace;
+        private final Timer.Context context;
+        private final Counter activeRequests;
+        private boolean ended = false;
+
+        public DropwizardContext(String metricNamespace) {
+            this.metricNamespace = Objects.requireNonNull(metricNamespace, "metricNamespace");
+            this.context = metricsRegistry.timer(name(metricNamespace, "requests", "latency")).time();
+            this.activeRequests = metricsRegistry.counter(name(metricNamespace, "requests", "active"));
+
+            activeRequests.inc();
+        }
+
+        @Override
+        public void endRequestMonitoring() {
+            if (!ended) {
+                ended = true;
+                activeRequests.dec();
+                context.stop();
+            } else {
+                LOGGER.warn("Request already ended");
```

Comment:
Would be a good idea to add the metric nameSpace to this warn so that we know what metric was "double ended."